woocommerce/google-listings-and-ads
# UX enhancement: Make the popover of the tooltip can be closed properly

### Changes proposed in this Pull Request:

It's a follow-up PR to solve the UX concern in https://github.com/woocommerce/google-listings-and-ads/pull/1894#issuecomment-1439586973:

> https://user-images.githubusercontent.com/5908855/220559751-dcf20af3-5e24-4e44-b75d-34cedea8915b.mov
> 
> Not sure if this is the targeted behaviour but I expected the tips to disappear. Didn't go deeper on it, but seems for me that when the mouse goes down the tip doesn't disappear correctly.

This PR implements one of the possible workarounds in GLA:

- ~Add a placeholder to `AppTooltip` for making the popover easier to be closed.~
- Forward the `disabled` prop of the actual child to the wrapper `<div>` in order to make `Tooltip` handle the close of tooltip properly.
- Extra tweak: Replace the CSS class prefix "app-" with "gla-" for the `AppTooltip` component as the `app-` might be too general to avoid a collision.

### Screenshots:

https://user-images.githubusercontent.com/17420811/220883326-183c5638-9ea8-436a-9fa8-45f2a1529a37.mp4

### Detailed test instructions:

💡 Considering the max limit of 20 is not that easy to test, it would be easier by changing the number in the following line to a smaller one:

https://github.com/woocommerce/google-listings-and-ads/blob/3ca5e71b58a863bfab520bd8143493ad3709374e/js/src/components/paid-ads/assetSpecs.js#L147

1. Go to step 2 of the campaign creation page.
2. Add the "Landscape", "Square" and "Portrait" images till reaching the max limit number to make the "Add image" buttons disabled.
3. Hover over a disabled button to view the popup.
4. Move out of the disabled button to see if it's easier to close the popup than before.
5. Repeat steps 3-4 but move the mouse out of the disabled button from different directions.

### Changelog entry

> Tweak - Make the popover of the tooltip can be closed properly.
